src/Cohesity.Powershell/Scripts/ProtectionSource/Register-CohesityProtectionSourceAWS.ps1

```diff
@@ -55,36 +55,78 @@ function Register-CohesityProtectionSourceAWS {
     [String]$FleetSubnetType,
     [Parameter(Mandatory = $false)]
     # Fleet settings, fleet tags.
-    [object[]]$FleetTags
+    [object[]]$FleetTags,
+    [Parameter(Mandatory = $false)]
+    # AWS account id.
+    [string]$AccountId,
+    [Parameter(Mandatory = $false)]
+    # IamRole registration name.
+    [string]$IamRoleName
   )
 
   Begin {
   }
 
   Process {
+    
+    if ($AuthMethod -eq "kUseIAMUser") {
+        $uri = '/irisservices/api/v1/public/protectionSources/register'
 
-    $uri = '/irisservices/api/v1/public/protectionSources/register'
-
-    $awsRegistrationParameters = @{
-      awsCredentials = @{
-        accessKey          = $AccessKey
-        amazonResourceName = $AmazonResourceName
-        authMethod         = $AuthMethod
-        awsType            = $AWSType
-        iamRoleArn         = $ARN
-        secretAccessKey    = $SecretAccessKey
-        subscriptionType   = $SubscriptionType
-      }
-      environment    = "kAWS"
+        $awsRegistrationParameters = @{
+          awsCredentials = @{
+            accessKey          = $AccessKey
+            amazonResourceName = $AmazonResourceName
+            authMethod         = $AuthMethod
+            awsType            = $AWSType
+            iamRoleArn         = $ARN
+            secretAccessKey    = $SecretAccessKey
+            subscriptionType   = $SubscriptionType
+          }
+          environment    = "kAWS"
+        }
+    } else {
+        # the auth method would be kUseIAMRole
+        $ENTITY_TYPE = 16
```

Comment:
Is there no public API for this ?

Reply:
there are no published public API to support IamRole authentication method